File: backend/src/order/objects/OrderChatMessage.scala
package delivery.order.objects

import delivery.domain.OrderId

final case class OrderChatMessage(
    id: String,
    orderId: OrderId,
    senderRole: OrderChatRole,
    peerRole: OrderChatRole,
    messageType: OrderChatMessageType,
    content: String,
    createdAt: String,
    readAt: Option[String] = None
)

File: frontend/src/objects/order/OrderChatMessage.ts
import type { OrderId } from '@/objects/shared/ids'
import type { OrderChatMessageType } from './OrderChatMessageType'
import type { OrderChatRole } from './OrderChatRole'

export type { OrderChatMessageType } from './OrderChatMessageType'
export type { OrderChatRole } from './OrderChatRole'

export interface OrderChatMessage {
  id: string
  orderId: OrderId
  senderRole: OrderChatRole
  peerRole: OrderChatRole
  messageType: OrderChatMessageType
  content: string
  createdAt: string
}
